File: src/cs/Schematron.Insight/Validation/ResultStatus.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Reflection;

namespace Schematron.Insight.Validation
{
    [Flags]
    public enum ResultStatus : int
    {
        [Display(Name = "None")]
        None = 0,
        [Display(Name = "Syntax Error")]
        SyntaxError = 1 << 1,
        [Display(Name = "Assert")]
        Assert = 1 << 2,
        [Display(Name = "Report")]
        Report = 1 << 3
    }
}

File: src/java/Schematron.Insight/src/main/validation/schematron/insight/validation/ResultStatus.java
package schematron.insight.validation;

import java.util.EnumSet;

public enum ResultStatus {

	None{
		@Override
		public String toString() { return "None"; }
		public int toInteger() { return 0; }
	},
	SyntaxError{
		@Override
		public String toString() { return "Syntax Error"; }
		public int toInteger() { return 1 << 0; }
	},
	Assert{
		@Override
		public String toString() { return "Assert"; }
		public int toInteger() { return 1 << 1; }
	},
	Report{
		@Override
		public String toString() { return "Report"; }
		public int toInteger() { return 1 << 2; }
	};
	abstract int toInteger();

	public boolean isContains(EnumSet<ResultStatus> target){
		return target.contains(this);
	}
}
